### ISSUE TITLE
[examples] separate examples in a new repo

### ISSUE THREAD

**Issue by vacary** [siconos/siconos#93]
I am wondering if it would be useful to separate examples in a new repo siconos/examples.

Firstly, it will help to split the packaging. Secondly, using git lfs in another repo, it would be possible to add extra files such as movies, pictures, large reference and data files and so on. Thirdly, it would also render more accessible the new python notebook that Franck is developing.

With a little work in md files and adding jekyll we can generate a nice example manual.

What do you think ?

**Comment by radarsat1** [comment]
I think we should keep the examples in the main repository because they are useful there, but we should remove the big files (e.g. ref files) and make them work with or without them.  We can have a separate repository or something for storing and showcasing examples with "generated" files (videos etc), but we can take some time to experiment with that, whether it is git lfs or something else.  I think the ref files however make the .tar.gz much bigger for not much gain.   We should have some _minimal_ ref files for basic tests, but they are overkill for the examples.

In summary, I think we should keep example code in the repository, but removing ref files, and also follow your suggestion of making an "example manual" as a separate project.

**Comment by vacary** [comment]
ok. I understand you point of view and I was also in favor to examples in the main repo. The question is "what is the usefulness of thee examples:
1- to unit and system test the functionalities of the code. In that case, we case build a system of unit testing and system testing as in a large of number of projects that are launched by make test. In our case, the examples are launched in a separate building project ...
2- to show though the examples how works siconos. In that case, ref files are useless and a little perturbating for the user. Furthermore, we are restricting by the comparisons with ref files and a lot of examples are useless from the pedagogical point of view.

Why not selecting a small number of examples with coarse ref files (less than 100 values) that covers the maximum of user functionalities in Siconos that we leave in a directory testing in the siconos repo and creating a full siconos-examples repo with ref files but with a lot of info for the end-user ?

**Comment by radarsat1** [comment]
I think that's a good approach.  Coarse ref files may be a good option, had not thought about that.  I think it's a good idea to distinguish between tests and examples.  Tests should test for degrading performance as well as API / ABI changes, etc.  Examples should be just that.. examples of how to code simulations using Siconos.  We should still compile and run examples during testing to ensure they are not broken, but they shouldn't be depended upon to check for degraded performance etc.

We could even have very limited set of "coarse" refs to keep the repository small and keep a larger battery of tests in a separate repository.  Normally I think it's ok for the repository to be (relatively) large, but in the case of Siconos we generally just tag and tar the repository instead of building a release tarball, so we don't exclude large files, and it becomes 60 MB just for refs.

Another possibility, although probably too complicated but I will say it anyway, is to have the automated test system actually build an old version of Siconos and run its tests to produce the refs on demand, allowing to check for divergence from previous versions without actually storing the refs anywhere permanently.  (It could be done once during construction of the test image...)

**Comment by vacary** [comment]
After a discussion with @fperignon, we think that it could be a good idea to 
- keep the "examples" directory in siconos with a possible a renaming into "test". In this directory, we can keep the minimal set of examples that test the maximum of functionalities of siconos.  we keep only the doxygen documentation in the source file and we reduce the size of the ref file. This test will launch if we add the component "test" if the CMakeLists.txt. The goal is to test the functionalities and to reduce the size of the .git (we will so a filter-branch on origin master for that)
- create a repo siconos-tutorial with all the actual examples. we remove the test w.r.t to the ref file. We add documentation, ipyhton notebook, scientific paper and pictures and videos if needed using git lfs.
The goal is to use this repo through the git interface or through the online jupiter server.

What do you think ?

**Comment by vacary** [comment]
The repository siconos-tutorials exists now.

I will start to copy the examples directory in it, rename examples in test_examples and start to enlighten it

ok ?

**Comment by xhub** [comment]
This has been done. Maybe rewrite the history with filter-branch to remove all the references to the examples?
